### ISSUE TITLE
use --always for git describe

### ISSUE THREAD

**Issue by igor-raits** [rpm-software-management/rpm-gitoverlay#2]
it will return shortcommit as a fallback, so in that case just set version to 0 and release should be set to `0.NUMgCOMMIT" where`NUM` is a number of commits since the beginning.

**Comment by igor-raits** [comment]
partially related to https://github.com/libgit2/libgit2/issues/3765

**Comment by igor-raits** [comment]
partially implemented in 378c080. Only one thing is still required to do is implement flag in libgit2 and use flag here.

**Comment by igor-raits** [comment]
commit c336df8aa76a30858315827ebcd1acf2ec00fcff

```
Author: Igor Gnatenko <i.gnatenko.brain@gmail.com>
Date:   Mon May 16 11:29:09 2016 +0200

    add support for aliases, better parsing ovl, refactoring

    Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>
```
